### PR TITLE
fix(path): replaced filepath methodes with pathutil's for remotepath

### PR DESCRIPTION
**Pull request by cnlangzi**
### Changes
-

### Fixes
- fixed filepath issue on --remotepath. filepath.Join/filepath.Split/filepath.Dir are replaced with pathutil's

### Tests
Tasks to complete before merging PR:
- [ ]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/gosdk/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [ ]  Merge your system tests PR to master AFTER merging this PR

### Associated PRs (Link as appropriate):
- zboxcli: https://github.com/0chain/zboxcli/pull/390

### CODE REVIEW

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | master  |
| gosdk  | fix/filepath_issues  |
| zboxcli  | fix/filepath_issues  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

File: core/pathutil/path.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+				if it != "" {
+					items = append(items, it)
+				}
+			}
+		}
+	}
+	if len(items) == 0 {
+		return ""
+	}
+	return "/" + strings.Join(items, "/")
```
Comment (peterlimg) [inline core/pathutil/path.go:59]: We'd better use relative path. User can call `Joins("/", "the path")` to get absolute path.

**Review by peterlimg — CHANGES_REQUESTED** [review]

**Review by peterlimg — APPROVED** [review]
LGTM

**Comment by service-0chain** [comment]
Manual system tests [cancelled] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | master  |
| gosdk  | fix/filepath_issues  |
| zboxcli  | fix/filepath_issues  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [success] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | master  |
| gosdk  | fix/filepath_issues  |
| zboxcli  | fix/filepath_issues  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Pull request merged**